tests/core/core_scope_test.py

```diff
@@ -160,5 +160,14 @@ def f(scope):
     b = root.child(f)()
     self.assertFalse(jnp.allclose(a, b))
 
+  def test_empty_col_error(self):
+    root = Scope({})
+    with self.assertRaises(errors.ScopeCollectionNotFound):
+      root.param('test', nn.initializers.zeros)
+    root = Scope({'params': {}})
+    with self.assertRaises(errors.ScopeCollectionNotFound):
+      root.param('test', nn.initializers.zeros)
```

Comment:
In theory this error should be raised for any collection, but you are only testing it for the collection `params`. Maybe you could add a test doing something like `root.variable('col', 'test', nn.initializers.zeros)`?